### PR TITLE
Enable caching using memcached

### PR DESCRIPTION
**Pull request by ASBishop**
Jira: [OSP-30046](https://issues.redhat.com//browse/OSP-30046)

### CODE REVIEW

**Comment by ASBishop** [comment]
/hold

Looks like I need to update the functional tests.

**Comment by ASBishop** [comment]
/retest

"Error from server (NotFound): deployments.apps "openstack-operator-controller-manager" not found"

**Comment by ASBishop** [comment]
/retest

failed to wait for the created cluster claim to become ready

**Comment by ASBishop** [comment]
/retest

5 tempest failures all due to 401 Unauthorized errors when deleting a share in the teardown phase. The tests themselves all passed.

**Comment by ASBishop** [comment]
/retest

Current analysis is a spurious timeout in one test triggers a cascade of false timeouts in other tests. There are ongoing upstream efforts to address these issues in tempest, and the current advice is "cross your fingers and try again."

**Comment by gouthampacha** [comment]
There's an issue here I think that's causing these failures:

a number of ["deferred" cleanups fail](https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/origin-ci-test/pr-logs/pull/openstack-k8s-operators_manila-operator/164/pull-ci-openstack-k8s-operators-manila-operator-main-manila-operator-build-deploy-tempest/1720159827474255872/artifacts/manila-operator-build-deploy-tempest/openstack-k8s-operators-tempest/artifacts/stestr_results.html); The way this works is that the tests request a project credential, and create a bunch of resources for multiple test cases to share... and these common resources are cleaned up at the end of the test class. The cleanup is done with the project credentials that were saved when the resource was created... Under normal circumstances, this is fine.. but, I _think_ that tokens are expiring by the time the tests come around to clean up.. 

[one of the failures](https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/origin-ci-test/pr-logs/pull/openstack-k8s-operators_manila-operator/164/pull-ci-openstack-k8s-operators-manila-operator-main-manila-operator-build-deploy-tempest/1720159827474255872/artifacts/manila-operator-build-deploy-tempest/openstack-k8s-operators-tempest/artifacts/tempest.log) indicates this from the tempest log:

```
2023-11-02 20:12:01.034 174 INFO tempest.lib.common.rest_client [-] Request (ShareSnapshotMetadataNegativeTest:tearDownClass): 401 DELETE https://manila-public-openstack.apps.oko-15.oooci.ccitredhat.com/v2/bffc542dde8b4d2a934dc2f6344708e6/shares/4b715c31-0237-4393-b522-fde3f16aa8c5 0.150s
2023-11-02 20:12:01.034 174 DEBUG tempest.lib.common.rest_client [-] Request - Headers: {'Content-Type': 'application/json', 'Accept': 'application/json', 'x-openstack-manila-api-version': '2.78', 'X-Auth-Token': '<omitted>'}
        Body: None
    Response - Headers: {'date': 'Thu, 02 Nov 2023 20:12:00 GMT', 'server': 'Apache', 'www-authenticate': 'Keystone uri="https://keystone-public-openstack.apps.oko-15.oooci.ccitredhat.com"', 'content-length': '114', 'content-type': 'application/json', 'set-cookie': '19c75a96fcae8ddcdc1109ace5ea16b4=6b523f58cd0e10dfdb8231e69cf2a94d; path=/; HttpOnly; Secure; SameSite=None', 'connection': 'close', 'status': '401', 'content-location': 'https://manila-public-openstack.apps.oko-15.oooci.ccitredhat.com/v2/bffc542dde8b4d2a934dc2f6344708e6/shares/4b715c31-0237-4393-b522-fde3f16aa8c5'}
        Body: b'{"error": {"code": 401, "title": "Unauthorized", "message": "The request you have made requires authentication."}}' _log_request_full /usr/lib/python3.9/site-packages/tempest/lib/common/rest_client.py:464
2023-11-02 20:12:01.035 174 ERROR tempest.test [-] teardown of resources failed: Unauthorized
Details: {'code': 401, 'title': 'Unauthorized', 'message': 'The request you have made requires authentication.'}: tempest.lib.exceptions.Unauthorized: Unauthorized
Details: {'code': 401, 'title': 'Unauthorized', 'message': 'The request you have made requires authentication.'}
2023-11-02 20:12:01.035 174 ERROR tempest.test Traceback (most recent call last):
2023-11-02 20:12:01.035 174 ERROR tempest.test   File "/usr/lib/python3.9/site-packages/tempest/test.py", line 231, in tearDownClass
2023-11-02 20:12:01.035 174 ERROR tempest.test     teardown()
2023-11-02 20:12:01.035 174 ERROR tempest.test   File "/usr/lib/python3.9/site-packages/manila_tempest_tests/tests/api/base.py", line 1201, in resource_cleanup
2023-11-02 20:12:01.035 174 ERROR tempest.test     super(BaseSharesMixedTest, cls).resource_cleanup()
2023-11-02 20:12:01.035 174 ERROR tempest.test   File "/usr/lib/python3.9/site-packages/manila_tempest_tests/tests/api/base.py", line 183, in resource_cleanup
2023-11-02 20:12:01.035 174 ERROR tempest.test     cls.clear_resources(cls.class_resources)
2023-11-02 20:12:01.035 174 ERROR tempest.test   File "/usr/lib/python3.9/site-packages/manila_tempest_tests/tests/api/base.py", line 863, in clear_resources
2023-11-02 20:12:01.035 174 ERROR tempest.test     client.delete_share(res_id)
2023-11-02 20:12:01.035 174 ERROR tempest.test   File "/usr/lib/python3.9/site-packages/manila_tempest_tests/services/share/v2/json/shares_client.py", line 355, in delete_share
2023-11-02 20:12:01.035 174 ERROR tempest.test     resp, body = self.delete(uri, version=version)
2023-11-02 20:12:01.035 174 ERROR tempest.test   File "/usr/lib/python3.9/site-packages/manila_tempest_tests/services/share/v2/json/shares_client.py", line 85, in delete
2023-11-02 20:12:01.035 174 ERROR tempest.test     resp, body = super(SharesV2Client, self).delete(url, headers=headers,
2023-11-02 20:12:01.035 174 ERROR tempest.test   File "/usr/lib/python3.9/site-packages/tempest/lib/common/rest_client.py", line 339, in delete
2023-11-02 20:12:01.035 174 ERROR tempest.test     return self.request('DELETE', url, extra_headers, headers, body)
2023-11-02 20:12:01.035 174 ERROR tempest.test   File "/usr/lib/python3.9/site-packages/tempest/lib/common/rest_client.py", line 742, in request
2023-11-02 20:12:01.035 174 ERROR tempest.test     self._error_checker(resp, resp_body)
2023-11-02 20:12:01.035 174 ERROR tempest.test   File "/usr/lib/python3.9/site-packages/tempest/lib/common/rest_client.py", line 842, in _error_checker
2023-11-02 20:12:01.035 174 ERROR tempest.test     raise exceptions.Unauthorized(resp_body, resp=resp)
2023-11-02 20:12:01.035 174 ERROR tempest.test tempest.lib.exceptions.Unauthorized: Unauthorized
2023-11-02 20:12:01.035 174 ERROR tempest.test Details: {'code': 401, 'title': 'Unauthorized', 'message': 'The request you have made requires authentication.'}
2023-11-02 20:12:01.035 174 ERROR tempest.test 
2023-11-02 20:12:01.304 174 INFO tempest.lib.common.rest_client [req-d5d6e751-4ade-4432-a58f-28c0bda67e96 req-d5d6e751-4ade-4432-a58f-28c0bda67e96 ] Request (ShareSnapshotMetadataNegativeTest:tearDownClass): 204 DELETE https://keystone-public-openstack.apps.oko-15.oooci.ccitredhat.com/v3/users/e089ee11f54c49d69c80d0d946249008 0.269s
2023-11-02 20:12:01.305 174 DEBUG tempest.lib.common.rest_client [req-d5d6e751-4ade-4432-a58f-28c0bda67e96 req-d5d6e751-4ade-4432-a58f-28c0bda67e96 ] Request - Headers: {'Content-Type': 'application/json', 'Accept': 'application/json', 'X-Auth-Token': '<omitted>'}
        Body: None
```

Digging into this; the test requested share creation at `2023-11-02 20:06:34.930`

```
2023-11-02 20:06:34.930 15 INFO manila.api.openstack.wsgi [None req-9a6e7e2d-625d-4028-bbc2-e81487e20c61 0e8a8e1b67504046a9f0258119b05f4e bffc542dde8b4d2a934dc2f6344708e6 - - default default] POST https://manila-public-openstack.apps.oko-15.oooci.ccitredhat.com/v2/bffc542dde8b4d2a934dc2f6344708e6/shares
```

and cleanup was attempted at `2023-11-02 20:12:01.035` - that's a difference of 05:26.105 (m:s) ... 
The [default TTL for the cache](https://docs.openstack.org/keystonemiddleware/latest/middlewarearchitecture.html#improving-response-time) is 300s (``token_cache_time``)... so manila API would need to re-validate the token; and that seems to be failing. Should we try changing this?

File: templates/manila/config/00-config.conf [hunk 1]
```diff
@@ -25,13 +25,15 @@ db_max_retries = -1
...
 auth_url = {{ .KeystoneInternalURL }}
+memcached_servers = {{ .MemcachedServersWithInet }}
 auth_type = password
 project_domain_name = Default
 user_domain_name = Default
 project_name = service
 username = {{ .ServiceUser }}
 password = {{ .ServicePassword }}
 interface = internal
+token_cache_time = 600
```
Comment (ASBishop) [inline templates/manila/config/00-config.conf:36]: WIP: This is just to test whether tempest tests pass when the token timeout is increased.

**Review by gouthampacha — COMMENTED** [review]
/retest-required

**Comment by gouthampacha** [comment]
Three auth failures on the last run that resulted in two cleanup failures in the last run.. re-running the job to see if these repeat.

```
2023-11-06 20:21:10.288 17 WARNING keystonemiddleware.auth_token [None req-12246e36-4743-4a36-91f2-ecbca90c274e 008a357059784e6ca91079bac852312d 561e0ebd3a2f47c2bf02ef8318158c39 - - default default] Authorization failed for token: keystonemiddleware.auth_token._exceptions.InvalidToken: Token authorization failed
```

**Comment by gouthampacha** [comment]
/retest-required

image pull error:

https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/origin-ci-test/pr-logs/pull/openstack-k8s-operators_manila-operator/164/pull-ci-openstack-k8s-operators-manila-operator-main-manila-operator-build-deploy-tempest/1721646297523425280/artifacts/manila-operator-build-deploy-tempest/openstack-k8s-operators-gather/artifacts/must-gather/quay-io-openstack-k8s-operators-openstack-must-gather-sha256-87cb35e74bf88873ba52c0886f7e63e40c72d7c935590f767c3f1e099f818f7a/namespaces/openstack-operators/pods/openstack-operator-controller-manager-59fcc68f8d-vqrxl/openstack-operator-controller-manager-59fcc68f8d-vqrxl-describe

File: templates/manila/config/00-config.conf [hunk 2]
```diff
@@ -25,6 +25,9 @@ db_max_retries = -1
 [keystone_authtoken]
 www_authenticate_uri={{ .KeystonePublicURL }}
 auth_url = {{ .KeystoneInternalURL }}
+# WIP - Test w/o memcache
+# memcached_servers = {{ .MemcachedServersWithInet }}
+# token_cache_time = 600
```
Comment (gouthampacha) [inline templates/manila/config/00-config.conf:30]: @ASBishop : we've made some job changes; could you uncomment these and retry?

**Review by fmount — APPROVED** [review]
/lgtm

**Pull request merged**